File: dinq-mobile/dinq/lib/features/dinq/search/domain/entities/Menu.dart
class Menu {
  final String id;
  final String restaurantId;
  final String branchId;
  final int version;
  final bool isPublished;
  final DateTime? publishedAt;
  final List<Tab> tabs;
  final DateTime createdAt;
  final DateTime updatedAt;
  final String updatedBy;
  final bool isDeleted;
  final int viewCount;

  Menu({
    required this.id,
    required this.restaurantId,
    required this.branchId,
    required this.version,
    required this.isPublished,
    this.publishedAt,
    required this.tabs,
    required this.createdAt,
    required this.updatedAt,
    required this.updatedBy,
    this.isDeleted = false,
    this.viewCount = 0,
  });
}

class Tab {
  final String id;
  final String menuId;
  final String name;
  final List<Category> categories;
  final bool isDeleted;

  Tab({
    required this.id,
    required this.menuId,
    required this.name,
    required this.categories,
    this.isDeleted = false,
  });
}

class Category {
  final String id;
  final String tabId;
  final String? name;
  final List<Item> items;

  Category({
    required this.id,
    required this.tabId,
    this.name,
    required this.items,
  });
}

class Item {
  final String id;
  final String name;
  final String slug;
  final String categoryId;
  final String? description;
  final List<String>? images;
  final double price;
  final String currency;
  final bool isAvailable;
  final List<String>? allergies;
  final List<String>? userImages;
  final int? calories;
  final List<String>? ingredients;
  final int? preparationTime;
  final dynamic howToEat;
  final DateTime createdAt;
  final DateTime updatedAt;
  final bool isDeleted;
  final int viewCount;
  final double? averageRating;
  final List<String>? reviewIds;

  Item({
    required this.id,
    required this.name,
    required this.slug,
    required this.categoryId,
    this.description,
    this.images,
    required this.price,
    this.currency = "ETB",
    this.isAvailable = true,
    this.allergies,
    this.userImages,
    this.calories,
    this.ingredients,
    this.preparationTime,
    this.howToEat,
    required this.createdAt,
    required this.updatedAt,
    this.isDeleted = false,
    this.viewCount = 0,
    this.averageRating,
    this.reviewIds,
  });
}

File: dineq-frontend/src/Types/menu.ts
export interface NutritionalInfo {
  calories: number;
  protein: number;
  carbs: number;
  fat: number;
}

export interface MenuItem {
  id: string;
  name: string;
  name_am?: string;
  slug: string;
  description?: string;
  description_am?: string;
  price: number;
  currency: string;
  allergies?: string[] | string;
  allergies_am?: string;
  tab_tags: string[];
  tab_tags_am: string[];
  nutritional_info?: NutritionalInfo;
  preparation_time?: number;
  how_to_eat?: string;
  how_to_eat_am?: string;
  created_at: string;
  updated_at: string;
  is_deleted: boolean;
  view_count: number;
  average_rating: number;
  image_url: string;
}

export interface Menu {
  id: string;
  name: string;
  slug: string;
  restaurant_id: string;
  version: number;
  is_published: boolean;
  published_at: string;
  items: MenuItem[];
  created_at: string; //
  updated_at: string;
}
